compiler/noirc_evaluator/src/acir/acir_variable.rs

```diff
@@ -541,6 +541,26 @@ impl<F: AcirField, B: BlackBoxFunctionSolver<F>> AcirContext<F, B> {
         Ok(())
     }
 
+    /// Constrains the `lhs` and `rhs` to be non-equal.
+    pub(crate) fn assert_neq_var(
+        &mut self,
+        lhs: AcirVar,
+        rhs: AcirVar,
+        assert_message: Option<AssertionPayload<F>>,
+    ) -> Result<(), RuntimeError> {
+        let diff_var = self.sub_var(lhs, rhs)?;
+
+        let one = self.add_constant(F::one());
+        let _ = self.inv_var(diff_var, one)?;
```

Comment:
So `one` is the `predicate`, which will be [multiplied with](https://github.com/noir-lang/noir/blob/81a3e8f3c39bf65d191d75eec81b34961904e80e/compiler/noirc_evaluator/src/acir/acir_variable.rs#L372) `(lhs-rhs) * (1/(lhs-rhs))` before we check that `(lhs-rhs) * (1/(lhs-rhs)) * predicate == predicate`. 

Just curious: if you're trying to minimise the number of operations, why not get rid of the multiplication with the predicate, and go straight to `assert_eq_var(should_be_one, one)`? I mean I see that `inv_var` is convenient, but yesterday the `add_mul` would have been correct, but did more than needed, so you preferred to call the push method directly.

Reply:
> why not get rid of the multiplication with the predicate, and go straight to assert_eq_var(should_be_one, one)

We short circuit on multiplications with a constant value of one so I thought that this would be clearest.

Reply:
The difference here in my mind is that we're going through a more general case which short circuits whereas the `add_mul` case didn't.